### PR TITLE
Configure FOSOAuthServer Bundle

### PR DESCRIPTION
**Pull request by nouran77**

### CODE REVIEW

**Comment by nouran77** [comment]
To test FOSOAuth Server Bundle
run these commands 
$ php bin/console doctrine:schema:update
$ php bin/console fos:oauth-server:create-client --redirect-uri="..." --grant-type="authorization_code"

File: app/AppKernel.php [hunk 1]
```diff
@@ -24,6 +24,8 @@ public function registerBundles()
             new FOS\RestBundle\FOSRestBundle(),
             new Nelmio\ApiDocBundle\NelmioApiDocBundle(),
             new Knp\Bundle\MenuBundle\KnpMenuBundle(),
+            new FOS\OAuthServerBundle\FOSOAuthServerBundle(),
+
```
Comment (ahmed-alnaqa) [inline app/AppKernel.php]: The additional empty lines always make conflicts in git, so remove it.

File: src/Todo/Bundle/UserBundle/Entity/AccessToken.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+use FOS\OAuthServerBundle\Entity\AccessToken as BaseAccessToken;
+use Doctrine\ORM\Mapping as ORM;
+use FOS\OAuthServerBundle\Model\ClientInterface;
+
+/**
+ * AccessToken
+ *
+ * @ORM\Entity
+ * @ORM\Table(name="access_token")
```
Comment (ahmed-alnaqa) [inline src/Todo/Bundle/UserBundle/Entity/AccessToken.php]: The table name always `plural`

File: src/Todo/Bundle/UserBundle/Entity/AuthCode.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+use FOS\OAuthServerBundle\Entity\AuthCode as BaseAuthCode;
+use Doctrine\ORM\Mapping as ORM;
+use FOS\OAuthServerBundle\Model\ClientInterface;
+
+/**
+ * AuthCode
+ *
+ * @ORM\Entity
+ * @ORM\Table(name="auth_code")
```
Comment (ahmed-alnaqa) [inline src/Todo/Bundle/UserBundle/Entity/AuthCode.php]: The table name always plural

File: src/Todo/Bundle/UserBundle/Entity/RefreshToken.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+use FOS\OAuthServerBundle\Entity\RefreshToken as BaseRefreshToken;
+use Doctrine\ORM\Mapping as ORM;
+use FOS\OAuthServerBundle\Model\ClientInterface;
+
+/**
+ * RefreshToken
+ *
+ * @ORM\Entity
+ * @ORM\Table(name="refresh_token")
```
Comment (ahmed-alnaqa) [inline src/Todo/Bundle/UserBundle/Entity/RefreshToken.php]: The table name always plural

**Review by ahmed-alnaqa — CHANGES_REQUESTED** [review]

**Review by ahmed-alnaqa — CHANGES_REQUESTED** [review]
:+1:

**Pull request merged**